pyhocon/config_parser.py

```diff
@@ -661,7 +661,8 @@ def resolve_substitutions(cls, config, accept_unresolved=False):
                         cache_values = cache.get(substitution)
                         if cache_values is None:
                             continue
-                    if not isinstance(resolved_value, ConfigValues):
+
+                    if resolved_value:
```

Comment:
Is there a reason for this specific check?

Reply:
After I've removed this check, the parsing succeeds in some cases, where it previously failed, but fails in some cases, where it succeeded previously.